core/src/sumeragi/mod.rs

```diff
@@ -188,42 +178,50 @@ impl Sumeragi {
         block_hash: HashOf<VersionedCommittedBlock>,
     ) -> Vec<VersionedCommittedBlock> {
         self.internal
-            .wsv
+            .wsv_for_public_use
             .lock()
             .unwrap()
             .blocks_after_hash(block_hash)
     }
 
     pub fn get_random_peer_for_block_sync(&self) -> Option<Peer> {
-        use rand::{prelude::SliceRandom, SeedableRng};
+        use rand::{prelude::SliceRandom, RngCore, SeedableRng};
 
         let rng = &mut rand::rngs::StdRng::from_entropy();
-        self.internal
-            .wsv
+        let mut vec = self
+            .internal
+            .current_online_peers
             .lock()
-            .unwrap()
-            .peers()
-            .choose(rng)
-            .cloned()
+            .expect("lock on online peers for get random peer")
+            .iter()
+            .map(|peer| Peer::new((*peer).clone()))
+            .collect::<Vec<Peer>>();
+        if vec.is_empty() {
+            return None;
+        }
+        vec.sort();
```

Comment:
Yes. I also like it because by being specific about what operations you are performing it's easier to modify this later. Say you wanted to select a group of 4 unique peers. The path from what I've written to this hypothetical version is shorter than from I had used `choose`. The more basic the language features you are using, the more things they can be used for. Specific functions like `choose` have only one very specific function, and if you wanted to continue that pattern you would also need `choose_multiple_overlapping` and `choose_multiple_unique` et cetera. Or what if you decided you wanted to try out gathering a list of 8 peers where a peer can be repeated at most 3 times. You get the point.